src/test/java/org/candlepin/pki/certs/X509CertificateBuilderTest.java

```diff
@@ -97,39 +97,62 @@ public void testCreateX509Certificate() throws Exception {
         X509CertificateHolder holder = new X509CertificateHolder(cert.getEncoded());
         Extensions bcExtensions = holder.getExtensions();
 
-        // KeyUsage extension incorrect
-        assertTrue(KeyUsage.fromExtensions(bcExtensions)
-            .hasUsages(KeyUsage.digitalSignature | KeyUsage.keyEncipherment | KeyUsage.dataEncipherment));
+        // Verify key usage extension is present and critical
+        Extension keyUsageExt = bcExtensions.getExtension(Extension.keyUsage);
+        assertNotNull(keyUsageExt);
+        assertTrue(keyUsageExt.isCritical());
+
+        KeyUsage keyUsage = KeyUsage.fromExtensions(bcExtensions);
+
+        assertTrue(keyUsage.hasUsages(KeyUsage.digitalSignature));
+        assertTrue(keyUsage.hasUsages(KeyUsage.keyEncipherment));
+        assertTrue(keyUsage.hasUsages(KeyUsage.dataEncipherment));
+
+        // Verify the extended key usage is present, non-critical, and is configured correctly
+        Extension exKeyUsageExt = bcExtensions.getExtension(Extension.extendedKeyUsage);
+        assertNotNull(exKeyUsageExt);
+        assertFalse(exKeyUsageExt.isCritical());
+
+        assertTrue(ExtendedKeyUsage.fromExtensions(bcExtensions)
+            .hasKeyPurposeId(KeyPurposeId.id_kp_clientAuth));
 
-        // ExtendedKeyUsage extension incorrect
-        assertTrue(ExtendedKeyUsage
-            .fromExtensions(bcExtensions).hasKeyPurposeId(KeyPurposeId.id_kp_clientAuth));
+        // Verify we aren't generating any CA certs with this builder
+        Extension basicConstraintExt = bcExtensions.getExtension(Extension.basicConstraints);
+        assertNotNull(basicConstraintExt);
 
-        // Basic constraints incorrectly identify this cert as a CA
         assertFalse(BasicConstraints.fromExtensions(bcExtensions).isCA());
 
+        //
```

Comment:
Empty comment to remove

Reply:
Fixed; filled in the comment line rather than removing it